### PR TITLE
feat: async mode

### PR DESCRIPTION
**Pull request by gurgunday**
Closes https://github.com/mcollina/fastify-html/issues/11

Lets us return streams that get sent progressively as the html is rendered

**Not finished, README will be updated and more tests will be added**

ghtml code: https://github.com/gurgunday/ghtml/blob/f3deee94641ed0fcdedf7ecb7e8c5438e5b7eee3/src/html.js#L144-L244

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -94,6 +98,62 @@ test('one level layout', async t => {
   }
 })
 
+test('one level layout (async)', async t => {
+  const app = fastify()
+  await app.register(fastifyHtml, { async: true })
+
+  app.setErrorHandler(function (error, req, reply) {
+    console.error(error)
+  })
```
Comment (mcollina) [inline test.js:107]: please remove

**Review by mcollina — COMMENTED** [review]
Can you please add a test with the layout and a streaming template?

Also simulate something common, such as a list.

**Comment by gurgunday** [comment]
Yeah, sorry, I made the PR during a class in a hurry after realizing it would just work to swap html with the generator version

Will add more tests and update docs now

**Comment by gurgunday** [comment]
Done! Do you think `async` is a good option name? It's kinda like an async mode, so I thought to keep it simple

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**